app/scripts/metamask-controller.js

```diff
@@ -2204,12 +2207,65 @@ export default class MetamaskController extends EventEmitter {
   }
 
   async getTokenStandardAndDetails(address, userAddress, tokenId) {
-    const details =
-      await this.assetsContractController.getTokenStandardAndDetails(
+    const { tokenList } = this.tokenListController.state;
+    const { tokens } = this.tokensController.state;
+
+    const staticTokenListDetails =
+      STATIC_MAINNET_TOKEN_LIST[address.toLowerCase()] || {};
+    const tokenListDetails = tokenList[address.toLowerCase()] || {};
+    const userDefinedTokenDetails =
+      tokens.find(({ address: _address }) =>
+        isEqualCaseInsensitive(_address, address),
+      ) || {};
+
+    const tokenDetails = {
+      ...staticTokenListDetails,
+      ...tokenListDetails,
+      ...userDefinedTokenDetails,
+    };
+    const tokenDetailsStandardIsisERC20 =
+      isEqualCaseInsensitive(tokenDetails.standard, TokenStandard.ERC20) ||
+      tokenDetails.erc20 === true;
+    const noEvidenceThatTokenIsAnNFT =
+      !isEqualCaseInsensitive(tokenDetails.standard, TokenStandard.ERC1155) &&
+      !isEqualCaseInsensitive(tokenDetails.standard, TokenStandard.ERC721) &&
+      !tokenDetails.erc721;
+    const otherThanStandardDetailsAreERC20Like =
+      tokenDetails.decimals !== undefined && tokenDetails.symbol;
```

Comment:
[nits]
- I don't think decimals would ever be a falsy but not undefined value (i.e. 0)?
- `otherThanStandardDetailsAreERC20Like` -> `otherDetailsAreERC20Like` (i think this is more legible, since it sounds like you're saying some details are standard and others are not. After a few seconds I got what you were saying but)
- and lastly I think `||` rather than `&&` since the presence of one is still evidence that is is ERC20 like even when the other is absent.
```suggestion
    const otherDetailsAreERC20Like =
      tokenDetails.decimals || tokenDetails.symbol;
```

Reply:
> I don't think decimals would ever be a falsy but not undefined value (i.e. 0)?

Decimals can be `0`. Here is an example of a recent bug report where users expect us to support such tokens https://github.com/MetaMask/metamask-extension/issues/17839

Reply:
<img width="1154" alt="Screenshot 2023-03-06 at 5 29 46 PM" src="https://user-images.githubusercontent.com/34557516/223280214-d05c38ec-ba24-4c39-b127-95ea0115a66e.png">

Interesting... doesn't look like this contract actually has a `decimals` value at all... I wonder if this is the fallback issue again

Reply:
hmmm no it seems to revert when I instantiate it with `ethers` and call `decimals()`

Reply:
I'll have to spin up a contract on a testnet that has no decimals getter/value to test what this [Mentors4EDU](https://github.com/Mentors4EDU) is claiming there. I'll do that tomorrow

Reply:
> it seems to revert when I instantiate it with ethers and call decimals()
this would explain why support for the token broke within the past few weeks, as we shipped the asset contract controller update in January

Reply:
So if the token in that github issue is manually imported, the user is forced to set the decimals to 0

https://user-images.githubusercontent.com/7499938/223395613-f24bd4ee-d7f6-4141-b166-3e593358c342.mp4

Reply:
And here are three tokens in the token list we fetch from API that have 0 decimals specified in their contract:

https://etherscan.io/token/0xdacd69347de42babfaecd09dc88958378780fb62
https://etherscan.io/token/0xa150db9b1fa65b44799d4dd949d922c0a33ee606
https://etherscan.io/token/0x08d32b0da63e2c3bcf8019c9c5d849d7a9d791e6

Reply:
Ok got it! What do you think about the proposed name change? And `&&` ->. `||`?

Reply:
name change sounds good

`&&` -> `||` would mean that a token with a symbol but no decimals would be "erc 20 like", but an NFT can have a symbol but not decimals, right?

Reply:
I suppose that's true. Both ERC721s and ERC20s could have a symbol but no decimals.

Reply:
Maybe you should just check for decimals? Since it's the only one that is potentially unique to ERC20s?